packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3950,123 +3950,39 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
   /// If the [attributedLabel] is not null, the [textDirection] must also not be null.
   RenderSemanticsAnnotations({
     RenderBox? child,
+    required SemanticsProperties properties,
     bool container = false,
     bool explicitChildNodes = false,
     bool excludeSemantics = false,
-    bool? enabled,
-    bool? checked,
-    bool? toggled,
-    bool? selected,
-    bool? button,
-    bool? slider,
-    bool? keyboardKey,
-    bool? link,
-    bool? header,
-    bool? textField,
-    bool? readOnly,
-    bool? focusable,
-    bool? focused,
-    bool? inMutuallyExclusiveGroup,
-    bool? obscured,
-    bool? multiline,
-    bool? scopesRoute,
-    bool? namesRoute,
-    bool? hidden,
-    bool? image,
-    bool? liveRegion,
-    int? maxValueLength,
-    int? currentValueLength,
     AttributedString? attributedLabel,
```

Comment:
These also exist on SmeanticsProperties, no? Why are those allowed to be passed in separately?

Reply:
They had slightly different logic where they get wrapped in AttributeString.  I was afraid that if I moved it, I might accidentally start allocating more than one AttributeString.  It was hard for me to prove that I wouldn't without storing that value somewhere.

Reply:
It would be fine to store that value on the RenderObject. It's just confusing that the API of the RO allowed this value to be passed in in two different ways and one just silently gets ignored.